### PR TITLE
Updating Bazel Rules and Go Version

### PR DESCRIPTION
**Pull request by chrislovecnm**
- updated Go rules and using Go 1.16
- updated Gazelle rules and matching gazelle version in kazel release
- updated rules_docker
- updated rules_k8s
- now loading gazelle dependencies after go dependencies
- updated documentation in WORKSPACE file

### CODE REVIEW

**Comment by rail** [comment]
This also fixes the macos issue, no more `sudo` \o/

**Review by keith-mcclellan — CHANGES_REQUESTED** [review]
the only real issue here is that something collapsed all the description fields in the CRD into one line and thats going to break the yaml linter that the openshift CI uses.  I wouldn't mind a slightly more verbose comment on line 9 of WORKSPACE too, but otherwise this looks good.

File: config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml [hunk 1]
```diff
@@ -40,38 +40,29 @@ spec:
         description: CrdbCluster is the CRD for the cockroachDB clusters API
         properties:
           apiVersion:
-            description: 'APIVersion defines the versioned schema of this representation
-              of an object. Servers should convert recognized schemas to the latest
-              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
```
[the remark this answers is not in the document]
Comment (chrislovecnm) [reply config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml:43]: It is auto generated 🤷🏻‍♂️

File: WORKSPACE [hunk 2]
```diff
@@ -1,9 +1,12 @@
-# gazelle:repository_macro hack/build/repos.bzl%go_repositories
 workspace(name = "com_github_coachroachdb_cockroach_operator")
 
+#####################################
+# Bazel macros for downloading deps #
+#####################################
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
 
+# TODO update
```
[the remark this answers is not in the document]
Comment (chrislovecnm) [reply WORKSPACE:9]: I missed removing the comment

File: config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (keith-mcclellan) [reply config/crd/bases/crdb.cockroachlabs.com_crdbclusters.yaml:43]: guess we'll see then, may have to come back to this later

**Review by keith-mcclellan — APPROVED** [review]
LGTM

**Pull request merged**